Dockerfile

```diff
@@ -1,9 +1,43 @@
-FROM ubuntu:22.04
-RUN apt-get update && apt-get upgrade --yes && apt-get install git python3.9 python3-pip --yes
-ADD . /code
-WORKDIR /code
-
-RUN pip install .
-RUN ape plugins install alchemy ens etherscan foundry hardhat infura ledger solidity template tokens trezor vyper
-RUN pip install eth-rlp==0.3.0
+#---------------------------------------------------------------------------------------------
+# Copyright (c) ApeWorX. All rights reserved.
+# Licensed under the Apache License. See LICENSE in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+ARG PYTHON_VERSION="3.9"
+FROM python:${PYTHON_VERSION}
+
+RUN apt-get update && apt-get upgrade --yes && apt-get install git
+
+# See http://label-schema.org for metadata schema
+# TODO: Add `build-date` and `version`
+LABEL maintainer="ApeWorX" \
+      org.label-schema.name="ape" \
+      org.label-schema.license="Apache" \
+      org.label-schema.description="Ape Ethereum Framework." \
+      org.label-schema.url="https://docs.apeworx.io/ape/stable/" \
+      org.label-schema.usage="https://docs.apeworx.io/ape/stable/userguides/quickstart.html#via-docker" \
+      org.label-schema.vcs-url="https://github.com/ApeWorX/ape" \
+      org.label-schema.docker.cmd="docker run --volume $HOME/.ape:/root/.ape --volume $HOME/.vvm:/root/.vvm --volume $HOME/.solcx:/root/.solcx --volume $PWD:/root/project --workdir /root/project apeworx/ape compile"
+
+RUN useradd --create-home --shell /bin/bash harambe
+WORKDIR /home/harambe
+COPY . .
+
+RUN pip install --upgrade pip \
+    && pip install --no-cache-dir . \
+    && ape plugins install \
+      alchemy \
+      ens \
+      etherscan \
+      foundry \
+      hardhat \
+      infura \
+      ledger \
+      solidity \
+      template \
+      tokens \
+      trezor \
+      vyper
```

Comment:
wonder if it should just be replaced with `pip install --no-cache-dir .[recommended-plugins]`

I know that it doesn't work by itself, but if it's the second command it might work

We were more explicit here because it was that (without the previous install) and thus was failing, but it would be great to control this all from one place

Reply:
I tried that and it didn't work unfortunately. Same issue.

Reply:
Additionally, I like the idea of possibly parametrizing this list as Dockerfile args so I can make a Starknet one easier.

Reply:
game for that. would be really slick to allow making a set of images that people can pull from with the right sets of plugins for their use cases

Reply:
I did think of a way to do both.
We put a `recommended_plugins.txt` file and have `setup.py` create its list from it.
Then, we allow the Dockerfile an `ARG plugins_list="./recommended_plugins.txt"` so you can give Dockerfile custom lists but by default it uses the same values from `pip install .'[recommended_plugins]'`

Reply:
Hmm, maybe we can read that file in to set the extra directly, so it's from one source (don't forget to add to project data files under `setup` as well)